internal/exec/vendor_utils.go

```
@@ -607,3 +465,152 @@ func processVendorImports(
+
+func shouldSkipSource(s *schema.AtmosVendorSource, component string, tags []string) bool {
+	// Skip if component or tags do not match
+	// If `--component` is specified, and it's not equal to this component, skip this component
+	// If `--tags` list is specified, and it does not contain any tags defined in this component, skip this component
+	return (component != "" && s.Component != component) || (len(tags) > 0 && len(lo.Intersect(tags, s.Tags)) == 0)
+}
+
+func determineSourceType(uri *string, vendorConfigFilePath string) (bool, bool, bool) {
+	// Determine if the URI is an OCI scheme, a local file, or remote
+	useOciScheme := strings.HasPrefix(*uri, "oci://")
+	if useOciScheme {
```

golangci-lint (1.62.2)

SA9003: empty branch

(staticcheck)